### DISCUSSION TITLE
Reset for inputs and errors not working. Please guide me.

### DISCUSSION THREAD

**Discussion by singhAmandeep007** [react-hook-form/react-hook-form#3729] [Q&A]
I am trying to call the function reset() to clean the the input values  and errors if any , but nothing happens. 
Can someone help?
Also I want some guidance on how can I make my code better(reusable).

Here is the link to the app.: [https://codesandbox.io/s/react-form-hooksemantic-ui-3vjbm?file=/src/SignupPage.js]

**Comment by bluebill1049** [comment]
you are using a custom register (register at `useEffect`), that input ref is not been registered. Maybe you should use Controller or useController for those inputs.
